packages/formik/src/Formik.tsx

```diff
@@ -739,8 +741,13 @@ export function useFormik<Values extends FormikValues = FormikValues>({
 
   const submitForm = useEventCallback(() => {
     dispatch({ type: 'SUBMIT_ATTEMPT' });
-    return validateFormWithHighPriority().then(
-      (combinedErrors: FormikErrors<Values>) => {
+    return validateFormWithHighPriority()
+      .catch(error => {
```

Comment:
the new callback is called both when the validation returns field errors, and also when the validation throws. this is debatable. should a throwing validation also call the callback, or is it rather a sign that the client forgot to implement proper error handling? also a thrown validation error would be catchable by `handleSubmit`, so the question is if "duplicate" handling by the new callback makes sense here.
also IF it makes sense, should the caught error also be returned as parameter to the callback?